tests/tar/pkg_tar_test.py

```diff
@@ -113,23 +108,20 @@ def test_strip_prefix_dot(self):
 
   def test_strip_files_dict(self):
     content = [
-        {'name': '.'},
         {'name': './not-etc'},
         {'name': './not-etc/mapped-filename.conf'},
```

Comment:
Help me understand this -- is this change only supposed to be removing the `.` entry from the tarfile?

Reading the commit message, I wasn't sure if this change would remove the `.` from the tarfile altogether or just remove that one entry.  Might be worth clarifying that point

Reply:
This change only removes the '.' entry from the tar and zip files. 
That never should have been been in an archive.

But the side effect of the change is also that a followup in pkg_tar can allow the leading './' be removed, without archive sticking it right back in again. That will be a big enough PR in itself, because it adds a layer of new tests.